src/game/common/rts/team.cpp

```diff
@@ -1492,3 +1496,713 @@ bool Team::Damage_Team_Members(float amount)
+            }
+        }
+    } else {
+        m_targetObjectID = OBJECT_UNK;
+    }
+}
+
+Object *Team::Get_Team_Target_Object()
+{
+    if (m_targetObjectID == OBJECT_UNK) {
+        return nullptr;
+    }
+
+    Object *obj = g_theGameLogic->Find_Object_By_ID(m_targetObjectID);
+
+    if (obj != nullptr && obj->Get_Status(OBJECT_STATUS_STEALTHED) && !obj->Get_Status(OBJECT_STATUS_DETECTED)
+        && !obj->Get_Status(OBJECT_STATUS_DISGUISED)) {
+        obj = nullptr;
+    }
+
+    if (obj && obj->Is_Effectively_Dead()) {
+        obj = nullptr;
+    }
+
+    if (obj && obj->Get_Contained_By() != nullptr) {
+        obj = nullptr;
+    }
+
+    if (obj && obj->Is_KindOf(KINDOF_AIRCRAFT)) {
+        obj = nullptr;
+    }
+
+    if (obj == nullptr) {
+        m_targetObjectID = OBJECT_UNK;
+    }
+
+    return obj;
+}
+
+void Team::Set_Override_Team_Relationship(unsigned int id, Relationship relationship)
+{
+    if (id != 0) {
+        m_teamRelations->m_relationships[id] = relationship;
+    }
+}
+
+void Team::Set_Override_Player_Relationship(int id, Relationship relationship)
+{
+    if (id != -1) {
+        m_playerRelations->m_relationships[id] = relationship;
+    }
+}
+
+void Team::Remove_Override_Player_Relationship(int id)
+{
+    if (!m_playerRelations->m_relationships.empty()) {
+        if (id == -1) {
+            m_playerRelations->m_relationships.clear();
+        } else {
+            auto it = m_playerRelations->m_relationships.find(id);
+
+            if (it != m_playerRelations->m_relationships.end()) {
+                m_playerRelations->m_relationships.erase(it);
+            }
+        }
+    }
+}
+
+bool Team::Is_Idle() const
+{
+    DLINK_ITERATOR<Object> it = Iterate_Team_Member_List();
+
+    while (!it.Done()) {
+        AIUpdateInterface *update = it.Cur()->Get_AI_Update_Interface();
+
+        if (update != nullptr) {
+            if (!it.Cur()->Is_Effectively_Dead() && !update->Is_Idle()) {
+                return false;
+            }
+        }
+
+        it.Advance();
+    }
+
+    return true;
+}
+
+void Team::Notify_Team_Of_Object_Death()
+{
+    const TeamTemplateInfo *info = m_proto->Get_Template_Info();
+
+    if (info != nullptr) {
+        if (!info->m_scriptOnUnitDestroyed.Is_Empty()) {
+            g_theScriptEngine->Run_Script(info->m_scriptOnUnitDestroyed, this);
+        }
+    }
+}
+
+bool Loco_Set_Matches(int lstm, unsigned int surface_bit_flags)
+{
+    return static_cast<unsigned char>((lstm & ((4 * (surface_bit_flags & 2)) | surface_bit_flags & 1))) != 0;
+}
+
+bool Team::Did_All_Enter(PolygonTrigger *poly, unsigned int surfaces) const
+{
+    if (!m_enteredOrExited) {
+        return false;
+    }
+
+    bool b1 = false;
+    bool b2 = false;
+    bool b3 = false;
+    DLINK_ITERATOR<Object> it = Iterate_Team_Member_List();
+
+    while (!it.Done()) {
+        AIUpdateInterface *update = it.Cur()->Get_AI_Update_Interface();
+
+        if (update != nullptr) {
```

Comment:
can clean this up with
```
        bool matches = false;
        if (update != nullptr) {
            matches = Loco_Set_Matches(update->Get_Locomotor_Set()->Get_Valid_Surfaces(), surfaces)) {
        } else {
            matches = Loco_Set_Matches(1, surfaces);
        }
        
        if (matches){
            if (!it.Cur()->Is_Effectively_Dead())
            ...
        }
```

Reply:
Would rather follow the original psuedo more closely on things like this.